plugins/govppmux/govpp_channel.go

```diff
@@ -61,13 +112,32 @@ func (r *govppRequestCtx) ReceiveReply(reply govppapi.Message) error {
 	return err
 }
 
-// SendRequest sends asynchronous request to the vpp and receives context used to receive reply.
-// Plugin govppmux allows to re-send retry which failed because of disconnected vpp, if enabled.
-func (c *goVppChan) SendRequest(request govppapi.Message) govppapi.RequestCtx {
-	sendRequest := c.Channel.SendRequest
+// SendMultiRequest sends asynchronous request to the vpp and receives context used to receive reply.
+func (c *goVppChan) SendMultiRequest(request govppapi.Message) govppapi.MultiRequestCtx {
+	startTime := time.Now()
+
+	sendMultiRequest := c.Channel.SendMultiRequest
 	// Send request now and wait for context
-	requestCtx := sendRequest(request)
+	requestCtx := sendMultiRequest(request)
 
 	// Return context with value and function which allows to send request again if needed
-	return &govppRequestCtx{requestCtx, sendRequest, request, c.retry}
+	return &govppMultirequestCtx{
+		requestCtx:  requestCtx,
+		sendRequest: sendMultiRequest,
+		requestMsg:  request,
+		stopwatch:   c.stopwatch,
+		started:     startTime,
+	}
+}
+
+// ReceiveReply handles request and returns error if occurred.
+func (r *govppMultirequestCtx) ReceiveReply(reply govppapi.Message) (bool, error) {
+	// Receive reply from original send
+	last, err := r.requestCtx.ReceiveReply(reply)
+	r.stopwatch.TimeLog(r.requestMsg.GetMessageName()).LogTimeEntry(time.Since(r.started))
```

Comment:
This will log multiple times per each reply each with increasing elapsed time.

Reply:
yes, that's the intent. Time is measured since the request is sent, till the reply is received. In the multi-request scenario, there are several replies, every with its own time entry.

Reply:
I think it should definitely log the time only when last reply is received. That way you can see how many real multi request there have been.